nixos/tests/user-activation-scripts.nix

```diff
@@ -0,0 +1,33 @@
+import ./make-test-python.nix ({ lib, ... }: {
+  name = "user-activation-scripts";
+  meta = with lib.maintainers; { maintainers = [ chkno ]; };
+
+  machine = {
+    system.userActivationScripts.foo = "mktemp ~/user-activation-ran.XXXXXX";
+    users.users.alice = {
+      initialPassword = "pass1";
+      isNormalUser = true;
+    };
+  };
+
+  testScript = ''
+    def verify_user_activation_run_count(n):
+        machine.succeed(
+            '[[ "$(find /home/alice/ -name user-activation-ran.\\* | wc -l)" == %s ]]' % n
+        )
+
+
+    machine.wait_for_unit("multi-user.target")
+    machine.wait_for_unit("getty@tty1.service")
+    machine.wait_until_tty_matches(1, "login: ")
+    machine.send_chars("alice\n")
+    machine.wait_until_tty_matches(1, "Password: ")
+    machine.send_chars("pass1\n")
+    machine.send_chars('echo login""ok\n')
```

Comment:
What's the purpose of the `""` here?

Reply:
The following line waits for `loginok` to appear in the console.  Running `echo loginok` would cause this to appear in the terminal from the terminal echo, not the shell actually executing the command.  By running `echo login""ok` instead, the terminal echo doesn't prematurely trigger the is-the-shell-actually-running check — the shell must actually run to do the quote removal.

I've changed this to an easier-to-quickly-understand file-touch test.  Thanks!

Reply:
Ah, that makes sense, got it. Thanks for the explanation (and for changing it to `touch` for easier understanding).